File: frontend/src/api/quality.js
import { createApiClient } from './http.js'

const api = createApiClient('/quality')

export default {
  summary() {
    return api.get('/summary').then((r) => r.data)
  },
  symbol(symbol) {
    return api.get(`/symbol/${symbol}`).then((r) => r.data)
  },
  conflicts(limit = 100) {
    return api.get('/conflicts', { params: { limit } }).then((r) => r.data)
  },
  missingBars(limit = 100) {
    return api.get('/missing-bars', { params: { limit } }).then((r) => r.data)
  }
}

File: backend/app/api/quality.py
from fastapi import APIRouter

from app.services.data_quality import (
    get_summary,
    get_symbol_quality,
    list_conflicts,
    list_missing_bars,
)

router = APIRouter()


@router.get("/summary")
async def quality_summary():
    return get_summary()


@router.get("/symbol/{symbol}")
async def quality_symbol(symbol: str):
    return get_symbol_quality(symbol)


@router.get("/conflicts")
async def quality_conflicts(limit: int = 100):
    return {"items": list_conflicts(limit)}


@router.get("/missing-bars")
async def quality_missing(limit: int = 100):
    return {"items": list_missing_bars(limit)}
